include/quic/utils.hpp

```diff
@@ -583,7 +619,7 @@ namespace std
             }
 
             auto h = hash<string_view>{}(addr_data);
-            h ^= hash<in_port_t>{}(port) + inverse_golden_ratio + (h << 6) + (h >> 2);
+            h ^= hash<decltype(port)>{}(port) + inverse_golden_ratio + (h << 6) + (h >> 2);
```

Comment:
Educational question: Previously the hash type was explicit to match the declaration of `port`. Why the change to `decltype`? Is this for platform compatibility?

Reply:
Because f-ing Windows has half-baked posix support for things like `in_port_t` that are supposed to be provided by aren't.